HoRNDIS.cpp

```diff
@@ -582,182 +417,297 @@ bool HoRNDIS::createNetworkInterface() {
 }
 
 /***** Interface enable and disable logic *****/
+HoRNDIS::ReentryLocker::ReentryLocker(IOCommandGate *inGate, bool &inGuard)
```

Comment:
I know that RAII is a C++ pattern ... but every time I see it, it still makes me squirm 😃

Reply:
Well, but isn't that nice that the destructor is called automatically for EVERY exit path, and you don't need to worry about forgetting to unlock it ;)